documentation/modules/managing/con-consumer-config-properties.adoc

```diff
@@ -250,6 +250,19 @@ A shorter interval between consecutive heartbeats allows for quicker detection o
 The heartbeat interval must be lower, usually by a third, than the session timeout. 
 Decreasing the heartbeat interval reduces the chance of accidental rebalancing, but more frequent heartbeats increases the overhead on broker resources.
 
+[source,env]
+----
+# ...
+max.poll.records=500 # <1>
+session.timeout.ms=45000 # <2>
+heartbeat.interval.ms=3000 # <3>
+# ...
+----
+<1> Sets the number of processed records returned from the consumer.
+<2> Adjust the heartbeat interval lower according to anticipated rebalances.
```

Comment:
We used the third person in 1)
Also this does not adjust the heartbeat interval, this is the session timeout. I'm not sure this is a great example overall as this keeps `max.poll.records` to the default value and lowers the session and heartbeat values. While this may detect consumer failures faster this could trigger rebalances more eagerly due to the lower margin for delays.

Reply:
I've adjusted the poll records down and increased the timeout and interval.